src/accelerate/utils.py

```diff
@@ -51,7 +51,36 @@ def is_sagemaker_available():
 OPTIMIZER_NAME = "optimizer"
 
 
-class RNGType(Enum):
+class MetaEnum(EnumMeta):
+    "A metaclass that adds the ability to check if `self` contains an item with the `in` operator"
+
+    def __contains__(cls, item):
+        try:
+            cls(item)
+        except ValueError:
+            return False
+        return True
+
+
+class BaseEnum(Enum, metaclass=MetaEnum):
```

Comment:
```suggestion
class BaseEnum(Enum, metaclass=EnumWithContains):
```
To go with the suggestion above.